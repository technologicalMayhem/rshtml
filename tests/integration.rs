use insta::assert_snapshot;
use rshtml::{convert_file, convert_directory};

#[test]
fn single_file() {
    let path_in = "./tests/input/single_file/index.html";

    let result = convert_file(path_in).unwrap();

    assert_snapshot!(result);
}

#[test]
fn multi_file() {
    let path_in = "./tests/input/multi_file";

    let result = convert_directory(path_in).unwrap();

    assert_snapshot!(result)
}