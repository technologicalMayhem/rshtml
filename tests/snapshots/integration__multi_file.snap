---
source: tests/integration.rs
expression: result
---
pub mod subpages {
    pub struct Page1;

    impl Page1 {
        pub fn render(self) -> String {
            format!("<p> This is subpage 1. I don't do anything special.</p>")
        }

        pub fn new() -> Self {
            Self {  }
        }
    }

    pub struct Page2 {
        date: String,
    }

    impl Page2 {
        pub fn render(self) -> String {
            let date = self.date;
            format!("<p> I am subpage 2. The current date is {date}.</p>")
        }

        pub fn new(date: String) -> Self {
            Self { date }
        }
    }
}

pub struct Index {
    page_name: String,
    cur_time: String,
}

impl Index {
    pub fn render(self) -> String {
        let page_name = self.page_name;
        let cur_time = self.cur_time;
        format!("<p> Welcome to {page_name}. It is currently {cur_time}.</p>")
    }

    pub fn new(page_name: String, cur_time: String) -> Self {
        Self { page_name, cur_time }
    }
}

pub struct About {
    author: String,
}

impl About {
    pub fn render(self) -> String {
        let author = self.author;
        format!("<p> This webpage was created by {author}</p>")
    }

    pub fn new(author: String) -> Self {
        Self { author }
    }
}

